lib/logstash/outputs/s3.rb

```diff
@@ -252,13 +269,36 @@ def restore_from_crashes
     end
   end
 
+  public 
+  def shouldcleanup(prefix)
+     return @empty_uploads[prefix] > @no_event_wait
+  end
+
   public
   def move_file_to_bucket(file)
+    
+    @logger.debug("S3: moving to bucket ", :file => file)
+
+    basepath = Pathname.new @temporary_directory
+    dirname = Pathname.new File.dirname(file)
+    prefixpath = dirname.relative_path_from basepath
+    prefix = prefixpath.to_s
```

Comment:
I dont see any use of prefixpath other that to set `prefix`
we can replace this with 

```ruby
prefix = dirname.relative_path_from(basepath).to_s
```